pipeline_steps/common.groovy

```diff
@@ -670,6 +670,12 @@ String clone_with_pr_refs(
 
 
 String clone_repo(String directory, String ssh_key, String repo, String ref, String refspec) {
+  // Need to clone/fetch with ssh@ protocol
+  if (ref.startsWith("https://")) {
+    beforeConversion = ref
+    ref = ref.replaceAll("https://", "git@")
```

Comment:
I don't think this transformation is complete. Git urls are `git@github.com:owner/repo` wheres http urls are `https://github.com/owner/repo`

Reply:
Good catch!